PYME/Analysis/points/ripleys.py

```diff
@@ -158,10 +239,27 @@ def ripleys_k(x, y, n_bins, bin_size, mask=None, bbox=None, z=None, threaded=Fal
     yu = yu.ravel()
     if zu is not None:
         zu = zu.ravel()
-            
-    return ripleys_k_from_mask_points(x=x, y=y, z=z,
+
+    bb, K = ripleys_k_from_mask_points(x=x, y=y, z=z,
                                       xu=xu, yu=yu, zu=zu,
                                       n_bins=n_bins, bin_size=bin_size, mask_area=mask_area, threaded=threaded)
+
+    if mc:
+        # Monte-Carlo simulations on the mask
+        K_arr = np.zeros((n_sim,len(K)))
+        for _i in range(n_sim):
+            xm, ym, zm, _ = mc_points_from_mask(mask, len(x), three_d, coord_origin)
+            _, K_arr[_i,:] = ripleys_k_from_mask_points(x=xm, y=ym, z=zm,
+                                      xu=xu, yu=yu, zu=zu,
+                                      n_bins=n_bins, bin_size=bin_size, mask_area=mask_area, threaded=threaded)
+        # Envelope
+        K_min = np.min(K_arr, axis=0)
+        K_max = np.max(K_arr, axis=0)
+        # Probability our original data is clustered
+        p_clustered = ((K_arr>=K).sum(0) + 1)/(n_sim + 1)
```

Comment:
What is the statistical basis of this test? Should be better documented here ...

Reply:
Based on the graphs, it's also clearly wrong as it should be impossible to get a p value of 1, and you should not even come close with 3 MC iterations.

Reply:
Will update the documentation! 

I think a p-value of 1 is fine from a mathematical standpoint, just maybe not meaningful for only 3 simulations. E.g.: `scipy.stats.ttest_ind([0.1,0.1,0.1],[0.1,0.1,0.1])` returns a p-value of 1.0.

Reply:
I'm pretty sure p is defined as strictly less than 1, but suspect that it doesn't matter that much because p values > .05 (or what ever your significance cutoff is) have no statistical interpretation (a p value ~=1 is not evidence for the opposite of what is being tested).

Reply:
Thought about this more. I think you're right a p-value should be strictly less than 1 for a continuous distribution. I think for a discrete and finite distribution, it should be possible to have p=1.